federation/apis/core/v1/doc.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is probably not required for now because we are reusing the types
+// in k8s.io/kubernetes/pkg/api. But adding it now as a guard to ensure
+// that others don't stumble on this in the future when they add new
+// types here since adding it now doesn't hurt.
+// +k8s:conversion-gen=k8s.io/kubernetes/federation/apis/core
```

Comment:
I am still trying to understand this fully, but I think its fine to remove it if its not required.
Up to you